sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -2202,8 +2216,25 @@ pi_result cuda_piextKernelSetArgMemObj(pi_kernel kernel, pi_uint32 arg_index,
 
   pi_result retErr = PI_SUCCESS;
   try {
-    CUdeviceptr cuPtr = (*arg_value)->get();
-    kernel->set_kernel_arg(arg_index, sizeof(CUdeviceptr), (void *)&cuPtr);
+    pi_mem arg_mem = *arg_value;
+    if (arg_mem->mem_type_ == _pi_mem::mem_type::surface) {
+      CUDA_ARRAY3D_DESCRIPTOR arrayDesc;
+      PI_CHECK_ERROR(cuArray3DGetDescriptor(
+          &arrayDesc, arg_mem->mem_.surface_mem_.get_array()));
+      if (arrayDesc.Format != CU_AD_FORMAT_UNSIGNED_INT32 &&
+          arrayDesc.Format != CU_AD_FORMAT_SIGNED_INT32 &&
+          arrayDesc.Format != CU_AD_FORMAT_HALF &&
+          arrayDesc.Format != CU_AD_FORMAT_FLOAT) {
+        cl::sycl::detail::pi::die(
```

Comment:
@nyalloc, sorry for raising such an old PR. Could you please comment why only these four types are supported for an image. Is that the CUDA limitation or something to be fixed on plugin side?

Reply:
hi @vladimirlaz I can answer to this, so the reason that just these image formats are supported is based on the interest in the feature at the time they were implemented, we're not aware of any limitation in CUDA that would prevent the remaining SYCL 2020 image formats (8-bit and 16-bit integer) being supported, though someone would have to investigate this.

Reply:
@AerialMantis thank you for the info